src/type_resolve_helpers.ts

```diff
@@ -4,15 +4,316 @@ import { warn } from './logger';
 import { PropTree, IPropDesc } from './PropTree';
 import { type } from 'os';
 
-const rgxObjectTokenize = /(<|>|,)/;
+const rgxObjectTokenize = /(<|>|,|\(|\)|\|)/;
 const rgxCommaAll = /,/g;
 const rgxParensAll = /\(|\)/g;
 
 const anyTypeNode = ts.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword);
 const strTypeNode = ts.createKeywordTypeNode(ts.SyntaxKind.StringKeyword);
 
-const anyGeneric: IGenericType = { kind: 'type', name: 'any', resolved: anyTypeNode };
-const strGeneric: IGenericType = { kind: 'type', name: 'string', resolved: strTypeNode };
+enum tsNodeTypes {
```

Comment:
nit: Please use cap-first camel case with an `E` prefix for enums. So `ENodeType`.